### PR TITLE
[WED-74] Gift request management/edit/create, etc

### PR DESCRIPTION
**Pull request by jonathanchiu**
This mainly implements functionality to manage gift requests from a therapist's perspective. Crap ton of smaller stuff sprinkled in here as well (font awesome, more RESTful routes, etc)

Also adds styling for edit/create pages for wishlists and requests so that stuff is prettier

### CODE REVIEW

File: app/controllers/gift_requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class GiftRequestsController < ApplicationController
+  def new
+    @gift_request = GiftRequest.new
+    render "gift_request", locals: {list_id: params[:id]}
```
Comment (ThePletch) [inline app/controllers/gift_requests_controller.rb]: have you tested this? my understanding of nested routing says that `lists/1/gift_request/new` should put the '1' in `params[:list_id]`.

File: spec/controllers/gift_requests_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  describe "GET #new" do
+    it "returns http success and renders template" do
+      get :new
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:gift_request)
+    end
+  end
+
+  describe "DELETE #destroy_multiple" do
+    it "successfully destroys a gift request record" do
```
Comment (ThePletch) [inline spec/controllers/gift_requests_controller_spec.rb]: should have a test to make sure it destroys multiple records when passed multiple IDs

**Comment by ThePletch** [comment]
Couple of minor points, otherwise looks great

**Comment by ThePletch** [comment]
Other thing: to work with the Amazon Product API, we should have a quick JS test that detects when an Amazon link is entered and tells them that we'll pull the product's name off Amazon for them.

File: app/controllers/gift_requests_controller.rb (hunk 1 above)
Comment (jonathanchiu) [inline app/controllers/gift_requests_controller.rb]: Yup I did, and I thought so too when I was writing this initially, but the log file shows:

```ruby

Started GET "/list/1/gift_requests/new" for ::1 at 2016-04-10 20:59:58 -0400
Processing by GiftRequestsController#new as HTML
  Parameters: {"id"=>"1"}
{"controller"=>"gift_requests", "action"=>"new", "id"=>"1"}
  Rendered gift_requests/gift_request.slim within layouts/application (21.1ms)
Completed 200 OK in 152ms (Views: 150.5ms | ActiveRecord: 0.0ms)
```

File: spec/controllers/gift_requests_controller_spec.rb (hunk 2 above)
Comment (jonathanchiu) [inline spec/controllers/gift_requests_controller_spec.rb]: Good call

File: app/controllers/gift_requests_controller.rb (hunk 1 above)
Comment (ThePletch) [inline app/controllers/gift_requests_controller.rb]: solid, okay

**Comment by ThePletch** [comment]
:sheep: it

**Comment by jonathanchiu** [comment]
Aye aye :rocket:

**Pull request merged**